test/model/deepstate/test_issm.py

```diff
@@ -0,0 +1,369 @@
+        features
+    )
+
+    assert (
+        (
+            emission_coeff
+            == mx.nd.ones((2, 10, issm.output_dim(), issm.latent_dim()))
+        )
+        .asnumpy()
+        .all()
+    )
+
+    assert (
+        (
+            transition_coeff
+            == mx.nd.ones((2, 10, issm.latent_dim(), issm.latent_dim()))
+        )
+        .asnumpy()
+        .all()
+    )
+
+    assert (
+        (innovation_coeff == mx.nd.ones((2, 10, issm.latent_dim())))
+        .asnumpy()
+        .all()
+    )
+
+
+def test_level_trend_issm_h():
+    issm = LevelTrendISSM()
+
+    assert issm.latent_dim() == 2
+    assert issm.output_dim() == 1
+
+    time_features = issm.time_features()
+
+    assert len(time_features) == 1
+
+    time_indices = [
+        pd.date_range("2020-01-01 21:00:00", periods=10, freq="H"),
+        pd.date_range("2020-01-31 22:00:00", periods=10, freq="H"),
+    ]
+
+    features = mx.nd.array(
+        np.stack(
+            [
+                np.stack([f(time_index) for f in time_features], axis=-1)
+                for time_index in time_indices
+            ],
+            axis=0,
+        )
+    )
+
+    emission_coeff, transition_coeff, innovation_coeff = issm.get_issm_coeff(
+        features
+    )
+
+    assert (
+        (
+            emission_coeff
+            == mx.nd.ones((2, 10, issm.output_dim(), issm.latent_dim()))
+        )
+        .asnumpy()
+        .all()
+    )
+
+    for item in range(2):
+        for time in range(10):
+            sliced_transition_coeff = mx.nd.slice(
+                transition_coeff,
+                begin=(item, time, None, None),
+                end=(item + 1, time + 1, None, None),
+            )
+            assert (
+                (
+                    sliced_transition_coeff
+                    == mx.nd.array([[1.0, 1.0], [0.0, 1.0]])
+                )
+                .asnumpy()
+                .all()
+            )
+
+    assert (
+        (innovation_coeff == mx.nd.ones((2, 10, issm.latent_dim())))
+        .asnumpy()
+        .all()
+    )
+
+
+def test_seasonality_issm_h():
+    issm = SeasonalityISSM(num_seasons=12, time_feature=MonthOfYearIndex())
+
+    assert issm.latent_dim() == 12
+    assert issm.output_dim() == 1
+
+    time_features = issm.time_features()
+
+    assert len(time_features) == 1
+
+    time_indices = [
+        pd.date_range("2019-01-01 21:00:00", periods=10, freq="H"),
+        pd.date_range("2019-01-31 22:00:00", periods=10, freq="H"),
+    ]
+
+    features = mx.nd.array(
+        np.stack(
+            [
+                np.stack([f(time_index) for f in time_features], axis=-1)
+                for time_index in time_indices
+            ],
+            axis=0,
+        )
+    )
+
+    emission_coeff, transition_coeff, innovation_coeff = issm.get_issm_coeff(
+        features
+    )
+
+    season_indices = [[0] * 10, [0] * 2 + [1] * 8]
+
+    for item in range(1, 2):
```

Comment:
You are checking here for only one item?

Reply:
right, that was for debugging, I'll revert it